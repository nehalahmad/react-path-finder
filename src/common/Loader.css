/**
 * Loader related CSS
 */
.loader-container {
  position: fixed;
  left: 0;
  top: 0;
  background-color: #80808038;
  width: 100%;
  height: 100%;
  z-index: 999;
}

.loader-container .loader {
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid #3498db;
  width: 120px;
  height: 120px;
  -webkit-animation: spin 2s linear infinite; /* Safari */
  animation: spin 2s linear infinite;
  position: absolute;
  left: calc(50% - 60px);
  top: calc(50% - 60px);
}

/* Safari */
@-webkit-keyframes spin {
  0% {
    -webkit-transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
