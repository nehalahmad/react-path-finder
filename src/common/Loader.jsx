// common/Loader.jsx
import React from "react";
import PropTypes from "prop-types";

import "./Loader.css";

/**
 * @description: Show loader when APi calls
 */
const Loader = ({ isLoading }) =>
  isLoading && (
    <div className="loader-container">
      <div className="loader" />
    </div>
  );

Loader.propTypes = {
  isLoading: PropTypes.bool.isRequired
};

export default Loader;
