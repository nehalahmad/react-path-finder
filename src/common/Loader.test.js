import React from "react";
import { shallow } from "enzyme";

import Loader from "./Loader";

describe("<Loader />", () => {
  it("running", () => {
    const LoaderComp = shallow(<Loader isLoading />);
    expect(LoaderComp).toMatchSnapshot();
  });

  it("not running", () => {
    const LoaderComp = shallow(<Loader isLoading={false} />);
    expect(LoaderComp).toMatchSnapshot();
  });
});
