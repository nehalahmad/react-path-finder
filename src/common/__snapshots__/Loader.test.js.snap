// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> not running 1`] = `ShallowWrapper {}`;

exports[`<Loader /> running 1`] = `ShallowWrapper {}`;
